File: backend/database/classes/skill.py
"""Classe Skill"""
from sqlalchemy import (
    Column,
    Integer,
    String,
    Text,
    Boolean,
)
from sqlalchemy.orm import relationship
from database.classes.base import SWdleBase


# Classe Skill
class Skill(SWdleBase):
    __tablename__ = 'skills'
    __table_args__ = {'extend_existing': True}

    name = Column(String(255), nullable=False)
    description = Column(Text)
    cooldown = Column(Integer)
    hits = Column(Integer)
    aoe = Column(Boolean)
    passive = Column(Boolean)
    slot = Column(Integer)
    image = Column(String(255))
    nb_progress = Column(Integer)

    monsters = relationship("MonsterSkill", back_populates="skill")

File: types/Skill.ts
// types/Skill.ts
export interface Skill {
    id: number;
    name: string;
    description?: string | null;
    cooldown?: number | null;
    hits?: number | null;
    aoe?: boolean | null;
    passive?: boolean | null;
    slot?: number | null;
    image?: string | null;
    nb_progress?: number | null;
}